rust/worker/src/segment/types.rs

```diff
@@ -26,6 +33,273 @@ impl<'a> MaterializedLogRecord<'a> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct MaterializedLogRecordV2<'a> {
```

Comment:
Any particular reason do this v1/v2? lets just replace the existing one?

Reply:
Yep, once I cleanup in the next PR, I will rename it